### PR TITLE
Update sidecars and driver to latest versions

### PR DESCRIPTION
**Pull request by msau42**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Updates the sidecars and driver to the latest versions.

Removes deploy-hostpath.sh as it is a duplicate of deploy.sh

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Add 1.18 deployment specs and update sidecar and driver versions.
```

### CODE REVIEW

**Comment by msau42** [comment]
/hold
for testing against 1.18

**Comment by msau42** [comment]
1.18 tests passed with prow override: https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/kubernetes-csi_csi-driver-host-path/169/pull-kubernetes-csi-csi-driver-host-path-1-17-on-kubernetes-1-17/1248421461106888704

**Comment by msau42** [comment]
Putting override here for reference:
```
CSI_PROW_KUBERNETES_VERSION="1.18.0"
CSI_PROW_KUBERNETES_DEPLOYMENT="1.18"
```

**Comment by msau42** [comment]
/assign @xing-yang @jsafrane 
I will be adding 1.18 jobs after this merges

File: deploy/kubernetes-1.18/README.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+The deployment for Kubernetes 1.18 uses VolumeSnapshot Beta CRDs and thus is imcompatible
```
Comment (xing-yang) [inline deploy/kubernetes-1.18/README.md:1]: s/imcompatible/incompatible

Comment (msau42) [reply deploy/kubernetes-1.18/README.md]: fixed

**Pull request merged**